parlai/tasks/ccpe/agents.py

```diff
@@ -3,150 +3,106 @@
-                        utterance['segments']
-                    ) if 'segments' in utterance else currSegments.append([])
-                    entry['segments'] = currSegments
-                    currEp.append(entry)
-                    entry = {}
-                    currSegments = []
-
-            flattenedData.append(currEp)
-
-        self.userData = []
-        self.assistantData = []
-
-        for ep in range(len(flattenedData)):
-            currUserEp = []
-            currAssistantEp = []
-
-            userCnt = 0
-            asssistantCnt = 0
-            for i, currUtt in enumerate(flattenedData[ep]):
-                if i > 0:
-                    if (
-                        currUtt['speaker'] == 'USER'
-                        and flattenedData[ep][i - 1]['speaker'] == 'ASSISTANT'
-                    ):
-                        entry = []
-                        entry.append(userCnt)
-                        entry.append(currUtt['text'])
-                        entry.append([flattenedData[ep][i - 1]['text']])
-                        entry.append(currUtt['segments'])
-                        entry.append(flattenedData[ep][i - 1]['segments'])
-                        entry.append(False)
-                        currUserEp.append(entry)
-                        userCnt += 1
-                    if (
-                        currUtt['speaker'] == 'ASSISTANT'
-                        and flattenedData[ep][i - 1]['speaker'] == 'USER'
-                    ):
-                        entry = []
-                        entry.append(asssistantCnt)
-                        entry.append(currUtt['text'])
-                        entry.append([flattenedData[ep][i - 1]['text']])
-                        entry.append(currUtt['segments'])
-                        entry.append(flattenedData[ep][i - 1]['segments'])
-                        entry.append(False)
-                        currAssistantEp.append(entry)
-                        asssistantCnt += 1
-
-            currUserEp[-1][5] = True
-            currAssistantEp[-1][5] = True
-            self.userData.append(currUserEp)
-            self.assistantData.append(currAssistantEp)
-
-        self.data = self.assistantData + self.userData
-
-    def get(self, episode_idx, entry_idx=0):
-        ep = self.data[episode_idx]
-        entry = ep[entry_idx]
-        action = {
-            'id': entry[0],
-            'text': entry[1],
-            'labels': entry[2],
-            'textSegments': entry[3],
-            'labelSegments': entry[4],
-            'episode_done': entry[5],
-        }
-        return action
-
-    def share(self):
-        shared = super().share()
-        shared['data'] = self.data
-        return shared
-
-
-class CCPEAssistantTeacher(CCPEAllTeacher):
-    def _setup_data(self):
-        super()._setup_data()
-        self.data = self.assistantData
-
-
-class DefaultTeacher(CCPEAllTeacher):
+        # do a 80:10:10 train/valid/test split
```

Comment:
should we do a random shuffle of the data here? or does it not matter

Reply:
The dataset doesn't specify a train/valid/test split.. I can see shuffling by a random seed making sense here though.